Sources/TuistLoader/Loaders/ManifestLoader.swift

```diff
@@ -245,7 +272,24 @@ public class ManifestLoader: ManifestLoading {
             let manifest = string[startTokenRange.upperBound ..< endTokenRange.lowerBound]
             return manifest.data(using: .utf8)!
         case let .failed(_, error):
+            handleUnexpectedImportError(in: path, error: error, manifest: manifest, plugins: plugins)
             throw error
         }
     }
+
+    /// Logs a help message to the user if attempting to import modules that
+    /// are disallowed from certain manifests.
+    private func handleUnexpectedImportError(in path: AbsolutePath, error: Error, manifest: Manifest, plugins: Plugins) {
+        guard case let TuistSupport.SystemError.terminated(command, _, standardError) = error,
+            manifest == .config || manifest == .plugin,
+            command == "swiftc",
+            let errorMessage = String(data: standardError, encoding: .utf8) else { return }
+
+        let defaultHelpersName = ProjectDescriptionHelpersBuilder.defaultHelpersName
+        let helperModules = [defaultHelpersName] + plugins.projectDescriptionHelpers.map(\.name)
+        let hasImportedUnexpectedModule = helperModules.reduce(false) { $0 || errorMessage.contains($1) }
+        guard hasImportedUnexpectedModule else { return }
+
+        logger.error("Can't import \(defaultHelpersName) or plugins in \(manifest.fileName(path))")
```

Comment:
This is an interesting function - and it probably makes sense to optimize this as it can happen quite often and it might not be immediately obvious what went wrong from the Swift default message. But in that case I'd test this in `ManifestLoaderIntegrationTests`, so we ensure that the contract between the Swift error message and the custom logic for handling the error is not broken.

Apart from that, I'd maybe divide the error in the default helpers and plugins. I think it's really important to know which one is actually missing and provide a custom error message for plugins with some pointers what the user could to fix the issue - as well as with default helpers (eg we could write something like: "Please check that you have defined project description helpers in `Tuist/ProjectDescriptionHelpers`")

Reply:
I've split the logs and added another one for build errors, think I may need some help here in general though as there are some edge cases that we should handle to have the best logging we can. I think this can be its own ticket